File: server/dotnet/Payment/CreatePaymentMapper.cs
using Microsoft.Extensions.Options;
using OnlinePayments.Sdk.Domain;

namespace OnlinePayments.Sdk.Example;

public class CreatePaymentMapper
{
    private readonly AppSettings _appSettings;

    public CreatePaymentMapper(IOptions<AppSettings> appSettings)
    {
        _appSettings = appSettings.Value;
    }

    /// <summary>
    /// Creates an empty create payment basic dto
    /// </summary>
    /// <returns>Instance of CreatePaymentBasicDto</returns>
    public CreatePaymentBasicDto ToEmptyBasicDto() {
        return new CreatePaymentBasicDto() {
            CardNumber = "4012000033330026",
            CardHolder = "Willie E. Coyote",
            ExpiryMonth = "05",
            ExpiryYear = "29",
            Cvv = "123"
        };
    }

    /// <summary>
    /// Converts CreatePaymentBasicDto to CreatePaymentRequest
    /// </summary>
    /// <param name="createPaymentBasicDto">Dto that needs to be converted</param>
    /// <returns>Instance of CreatePaymentRequest</returns>
    public CreatePaymentRequest ToCreatePaymentRequest(CreatePaymentBasicDto createPaymentBasicDto) {
        return new CreatePaymentRequest() {
            CardPaymentMethodSpecificInput = new CardPaymentMethodSpecificInput() {
                Card = new Card() {
                    CardNumber = createPaymentBasicDto.CardNumber,
                    CardholderName = createPaymentBasicDto.CardHolder,
                    ExpiryDate = string.Format("{0}{1}", createPaymentBasicDto.ExpiryMonth, createPaymentBasicDto.ExpiryYear),
                    Cvv = createPaymentBasicDto.Cvv
                },
                PaymentProductId = 1
            },
            Order = new Order() {
                AmountOfMoney = new AmountOfMoney() {
                    Amount = ToAmount(createPaymentBasicDto.Amount),
                    CurrencyCode = createPaymentBasicDto.Currency
                },
                Customer = new Customer() {
                    Device = new CustomerDevice() {
                        AcceptHeader = "text/html,application/xhtml+xml,application/xml;q=0.9,/;q=0.8",
                        BrowserData = new BrowserData() {
                            ColorDepth = 24,
                            JavaEnabled = false,
                            ScreenHeight = "1200",
                            ScreenWidth = "1920"
                        },
                        IpAddress = "123.123.123.123",
                        Locale = "en_GB",
                        UserAgent = "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_4) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/12.1 Safari/605.1.15",
                        TimezoneOffsetUtcMinutes = "420"
                    }
                }
            }
        };
    }

    private long ToAmount(decimal amount)
    {
        decimal multipliedAmount = decimal.Multiply(amount, new decimal(100));
        decimal roundedAmount = decimal.Round(multipliedAmount, 0, MidpointRounding.ToEven);
        return (long)roundedAmount;
    }
}

File: server/ruby/app/payment/create_payment_mapper.rb
require_relative '../services/payment_utils'

require 'onlinepayments/sdk'
include OnlinePayments::SDK::Domain

class CreatePaymentMapper
  def self.to_empty_dto
    CreatePaymentBasicDto.new(
      "4012000033330026",
      "Willie E. Coyote",
      "05",
      "29",
      "123",
      nil,
      nil
    )
  end

  def self.to_create_payment_request(create_payment_basic_dto)
    create_payment_request = CreatePaymentRequest.new

    card = Card.new
    card.card_number = create_payment_basic_dto.card_number
    card.cardholder_name = create_payment_basic_dto.card_holder
    card.expiry_date = "#{create_payment_basic_dto.expiry_month}#{create_payment_basic_dto.expiry_year}"
    card.cvv = create_payment_basic_dto.cvv

    card_payment_method_specific_input = CardPaymentMethodSpecificInput.new
    card_payment_method_specific_input.card = card
    card_payment_method_specific_input.payment_product_id = 1

    order = Order.new

    amount_of_money = AmountOfMoney.new
    amount_of_money.amount = PaymentUtils.to_amount(create_payment_basic_dto.amount)
    amount_of_money.currency_code = create_payment_basic_dto.currency

    order.amount_of_money = amount_of_money

    customer = Customer.new

    device = CustomerDevice.new
    device.accept_header = "text/html,application/xhtml+xml,application/xml;q=0.9,/;q=0.8"

    browser_data = BrowserData.new
    browser_data.color_depth = 24
    browser_data.java_enabled = false
    browser_data.screen_height = "1200"
    browser_data.screen_width = "1920"

    device.browser_data = browser_data
    device.ip_address = "123.123.123.123"
    device.locale = "en_GB"
    device.user_agent = "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_4) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/12.1 Safari/605.1.15"
    device.timezone_offset_utc_minutes = "420"

    customer.device = device

    order.customer = customer

    create_payment_request.order = order
    create_payment_request.card_payment_method_specific_input = card_payment_method_specific_input

    return create_payment_request

  end
end
